### PR TITLE
Single video page

### PR DESCRIPTION
**Pull request by mansipadgelwar**
Single video page
1. Added chips for categories
     -- If I click on any one of the categories I will be redirected to the video list page with the selected category.
2. Video Listing Page
     -- I can see a video listing page where all the videos are listed. 
     -- On the Video Cards, I can see a CTA button
3. Like a video
     -- I can like the video if I have logged in

Note: Dislike, add to watch later, and add to playlist from single video page is not implemented yet

### CODE REVIEW

File: src/components/CategoryChips/CategoryChips.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    (async () => {
+      try {
+        const response = await axios.get("/api/categories");
+        if (response.status === 200) {
+          const data = response.data.categories;
+          setCategory(data);
+          return data;
+        }
+      } catch (error) {
+        showToast(`Error fetching categories`, "error");
```
Comment (Shah-Arsalan) [inline src/components/CategoryChips/CategoryChips.jsx:24]: Great you have used toast for error!

File: src/components/modals/playlist-modal/playlist-modal.css [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  right: 0;
+  background: grey;
+  z-index: 1;
+}
+
+.modal {
+  position: relative;
+  max-width: 300px;
+  min-height: 200px;
+  background: white;
```
Comment (Shah-Arsalan) [inline src/components/modals/playlist-modal/playlist-modal.css]: use color variables

**Review by Shah-Arsalan — APPROVED** [review]
Code looks good . Your deploy doesn't look right .Can you check?

**Comment by mansipadgelwar** [comment]
> Code looks good . Your deploy doesn't look right .Can you check?

File: src/components/CategoryChips/CategoryChips.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const [categoryType, setCategory] = useState([]);
+  const { clickedCategory, setClickedCategory } = useData();
+  const { showToast } = useToast();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/categories");
+        if (response.status === 200) {
+          const data = response.data.categories;
```
Comment (dev-enforced) [inline src/components/CategoryChips/CategoryChips.jsx]: Can rewrite it using object destructuring:
```
const {data:{categories}}=response;
setCategory(categories)
```
Helps in accessing nested level data too.

File: src/services/index.js [hunk 4]
```diff
@@ -1,2 +1,16 @@
...
+export { deletePlaylistService } from "./playlist-services/deletePlaylistService";
+export { deleteVideoFromPlaylistOfUserService } from "./playlist-services/deleteVideoFromPlaylistOfUserService";
+export { getAllPlaylistOfUserService } from "./playlist-services/getAllPlaylistOfUserService";
+export { getPlaylistOfUserService } from "./playlist-services/getPlaylistOfUserService";
+export { addVideoToHistoryOfUserService } from "./history-services/addVideoToHistoryOfUserService";
+export { clearCompleteHistoryOfUserService } from "./history-services/clearCompleteHistoryOfUserService";
+export { deleteVideoFromHistoryOfUserService } from "./history-services/deleteVideoFromHistoryOfUserService";
+export { getHistoryOfUserService } from "./history-services/getHistoryOfUserService";
+export { deleteVideoFromWatchLaterService } from "./watch-later-services/deleteVideoFromWatchLaterService";
+export { getWatchLaterVideoOfUserService } from "./watch-later-services/getWatchLaterVideoOfUserService";
```
Comment (dev-enforced) [inline src/services/index.js]: Good use of barrel exports. Can further reduce it too later on by shifting these lines in a index.js file of each of the folders of history-services,watch-later-services,playlist-services.
And then by using those index.js of the folders while writing this file:
```js
src/services/index.js
export * from "./watch-later-services";
export * from "./history-services";
export * from "./playlist-services"
```

```js
//INDIVIDUAL FOLDER HAVING AN INDEX.JS file
//src/services/like-services/index.js
export { addVideoToWatchLaterService } from "./addVideoToWatchLaterService";

```

**Review by dev-enforced — APPROVED** [review]
The code structure looks really good!! The live preview is also working with all the single video views working in a proper manner https://deploy-preview-5--midori-video-library.netlify.app/videopage/oMpgga7ve7Y

Good work @mansipadgelwar on developing this responsive single video page view!!
image
Is there a need to have a dislike button too you can either like a video or remove a liked video for which you can alter the icon and text with a removeVideoFromLikedSection action handler on that basis?

**Pull request merged**